### PR TITLE
DBZ-3822 Add comments to handle commit message check on failure

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-3822

### CODE REVIEW

**Comment by ani-sha** [comment]
@gunnarmorling As requested here are the rest. Could you please review the wordings?

File: .github/workflows/sanity-check.yml [hunk 1]
```diff
@@ -16,3 +15,10 @@ jobs:
           BASE_REF: ${{ github.base_ref }}
         run: |
           ! git log --format='format:%s' refs/remotes/origin/$BASE_REF.. | tail -n +2 | grep -Ev '^(DBZ-[[:digit:]]+)|(\[release\])|(\[jenkins-jobs\]|(\[docs\]) '
+      - name: Create Comment
+        if: ${{ failure() }}
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            Hi @${{ github.event.pull_request.user.login }}. Please prefix the commit message(s) with the [DBZ-xxx JIRA issue key](https://github.com/debezium/debezium/blob/master/CONTRIBUTE.md#making-changes).
```
Comment (gunnarmorling) [inline .github/workflows/sanity-check.yml:24]:
```suggestion
            Hi @${{ github.event.pull_request.user.login }}, thanks for your contribution. Please prefix the commit message(s) with the [DBZ-xxx JIRA issue key](https://github.com/debezium/debezium/blob/master/CONTRIBUTE.md#making-changes).
```

File: .github/workflows/formatting-check.yml [hunk 2]
```diff
@@ -22,3 +22,12 @@ jobs:
...
     - name: Run formatting and import order checks
       run: mvn clean install -DskipTests=true -DskipITs=true -Dformat.formatter.goal=validate -Dformat.imports.goal=check -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.httpconnectionManager.ttlSeconds=120
+
+    - name: Create Comment
+      if: ${{ failure() }}
+      uses: peter-evans/create-or-update-comment@v1
+      with:
+        issue-number: ${{ github.event.pull_request.number }}
+        body: |
+          Hi @${{ github.event.pull_request.user.login }}. To validate the [formatting and import order checks](https://github.com/debezium/debezium/blob/master/CONTRIBUTE.md#code-formatting) please run the following command:
```
Comment (gunnarmorling) [inline .github/workflows/formatting-check.yml:32]:
```suggestion
          Hi @${{ github.event.pull_request.user.login }}, thanks for your contribution. In order to ensure a consistent [formatting and import ordering](https://github.com/debezium/debezium/blob/master/CONTRIBUTE.md#code-formatting), please run the following command and update this PR with the resulting changes:
```

File: .github/workflows/formatting-check.yml [hunk 3]
```diff
@@ -22,3 +22,12 @@ jobs:
...
+          `mvn clean install -DskipTests=true -DskipITs=true -Dformat.formatter.goal=validate -Dformat.imports.goal=check`
```
Comment (gunnarmorling) [inline .github/workflows/formatting-check.yml:33]:
```suggestion
          `mvn process-sources`
```

Here we want to actually apply folks the formatter, not just the checker.

**Review by gunnarmorling — APPROVED** [review]
LGTM, some minor suggestions inline.

**Comment by ani-sha** [comment]
Hey @gunnarmorling, I went through the concerns regarding `pull_request_target` as mentioned in Gitter. We can remove the checkout action from the `commit message check` as there is no need of checking out. But in the case of  `formatting check` we can't remove the checkout step. So what do you suggest?

**Comment by gunnarmorling** [comment]
@ani-sha, let's skip the change for the formatting check then. While it would be nice, I think we're better off to not do it in this case.

File: .github/workflows/formatting-check.yml [hunk 4]
```diff
@@ -6,19 +6,19 @@ jobs:
...
+      - uses: actions/checkout@v2
 
-    - name: Set up JDK
-      uses: actions/setup-java@v1.4.3
-      with:
-        java-version: "14"
+      - name: Set up JDK
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: "14"
```
Comment (gunnarmorling) [inline .github/workflows/formatting-check.yml:14]: As I'm seeing this, 14 is EOL by now. We should either be using 16 (latest one, till tomorrow, when 17 is out ;), or 11, the previous LTS. I suppose there will be issues with 16 due to the changes there in regards to reflective access. Can you capture this in a Jira issue?

Comment (ani-sha) [reply .github/workflows/formatting-check.yml:14]: @gunnarmorling Logged https://issues.redhat.com/browse/DBZ-3993 for upgrading the java version.

**Comment by gunnarmorling** [comment]
Applied; thx, @ani-sha!

**Pull request merged**